test/asset.test.js

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/* eslint-env mocha */
+/* eslint mocha/no-mocha-arrows: "off" */
+
+'use strict';
+
+const assert = require('assert');
+
+const Asset = require('../lib/asset');
+
+describe("asset.js", () => {
+    it('verifies constructor works properly', function () {
+        let asset = new Asset({}, '', 'file.png');
+        assert.ok(asset instanceof Asset);
+        assert.ok(asset.path, "./file.png");
+        assert.ok(asset.name, 'file.png');
+        assert.ok(asset.extension, 'png');
+        assert.strictEqual(asset.url, undefined);
+        assert.strictEqual(asset.size, undefined);
+        assert.strictEqual(asset.headers, undefined);
+        assert.strictEqual(asset.type, undefined);
+
+        // custom directory
+        asset = new Asset({}, '/path', 'file.png');
+        assert.ok(asset instanceof Asset);
+        assert.ok(asset.path, "/path/file.png");
+        assert.ok(asset.name, 'file.png');
+        assert.ok(asset.extension, 'png');
+        
+        // no extension
+        asset = new Asset({}, '/path', 'file');
+        assert.ok(asset instanceof Asset);
+        assert.ok(asset.path, "/path/file");
+        assert.ok(asset.name, 'file');
+        assert.ok(asset.extension, '');
+    });
+    it('verifies constructor works properly without params', function () {
```

Comment:
Could you add empty lines between tests? Files without newlines between tests are harder to read...